File: Go/Kex.go
package main

import (
	"fmt"
	"math/rand"
	"strconv"
)

const PlayerHp int = 500
const PlayerDefense int = 30

type Kex struct {

}

func (kex Kex) Execute() {
	fmt.Println("冒険が今始まる！")
	player := Player{
		Hp:      PlayerHp,
		Defense: PlayerDefense,
	}
	continueGame := 1
	kills := 0

	for continueGame != 0 {
		fmt.Printf("現HP：%d\n", player.Hp)
		fmt.Print("奥に進みますか？（１：奥に進む　０．帰る）＞")
		var reader string
		_, _ = fmt.Scanln(&reader)
		continueGame, _ = strconv.Atoi(reader)
		if continueGame != 0 {
			golemLv := rand.Intn(5)
			AttackGolemEx(golemLv, &player)
			kills++
			if player.Hp == 0 {
				break
			}
		}
	}
	fmt.Println("リ〇ミト！")
	fmt.Printf("戦闘回数：%d　残りHP：%d\n", kills, player.Hp)
}

File: SwiftProject/SwiftProject/Kex.swift
//
//  Kex.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/05/24.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

public class Kex {
    public static func Execute() {
        print("冒険が今始まる！")
        var player = Player()
        var continue_game = 1
        var kills = 0
        while continue_game != 0 {
            print("現HP：\(player.Hp)")
            print("奥に進みますか？（１．奥に進む　２．帰る）")
            continue_game = Int(readLine()!)!
            if continue_game != 0 {
                let golem_lv = Int.random(in: 0...4)
                AttackGolemEx(lv: golem_lv, player: &player)
                kills += 1
                if player.Hp == 0 {
                    break
                }
            }
        }
        print("リ◯ミト！")
        print("戦闘回数：\(kills)回　残りHP：\(player.Hp)")
    }
}
